internal/cmdcommon/extension.go

```diff
@@ -57,17 +52,47 @@ func (kec *KymaExtensionsConfig) GetRawExtensions() ExtensionList {
 	return kec.extensions
 }
 
-func (kec *KymaExtensionsConfig) BuildExtensions(availableTemplateCommands *TemplateCommandsList, availableCoreCommands CoreCommandsMap) []*cobra.Command {
-	cmds := make([]*cobra.Command, len(kec.kymaConfig.extensions))
+func (kec *KymaExtensionsConfig) BuildExtensions(availableTemplateCommands *TemplateCommandsList, availableCoreCommands CoreCommandsMap, cmd *cobra.Command, config *KymaConfig) []*cobra.Command {
+	var cmds []*cobra.Command
+
+	var cms, cmsError = getExtensionConfigMaps(config.Ctx, config.KubeClientConfig)
+	if cmsError != nil {
+		kec.parseErrors = cmsError
+		return nil
+	}
+
+	existingCommands := make(map[string]bool)
+	for _, baseCmd := range cmd.Commands() {
+		existingCommands[baseCmd.Name()] = true
+	}
+
+	for _, cm := range cms.Items {
+		extension, _ := parseResourceExtension(cm.Data)
+		if existingCommands[extension.RootCommand.Name] {
+			kec.parseErrors = errors.Join(
+				kec.parseErrors,
+				fmt.Errorf("failed to validate configmap '%s/%s': base command with rootCommand.name'%s' already exists",
```

Comment:
I would suggest the new format:
```suggestion
				fmt.Errorf("failed to validate configmap '%s/%s': base command with name='%s' already exists",
```